utils/tinystr/src/ascii.rs

```diff
@@ -19,6 +19,36 @@ impl<const N: usize> TinyAsciiStr<N> {
         Self::from_bytes_inner(bytes, 0, bytes.len(), false)
     }
 
+    /// Attempts to parse a fixed-length byte array to a `TinyAsciiStr`.
+    ///
+    /// The byte array may contain trailing NUL bytes.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// use tinystr::TinyAsciiStr;
+    /// use tinystr::tinystr;
+    ///
+    /// assert_eq!(
+    ///     TinyAsciiStr::<3>::try_from_raw(b"GB\0"),
+    ///     Ok(&tinystr!(3, "GB")));
+    /// assert_eq!(
+    ///     TinyAsciiStr::<3>::try_from_raw(b"USD"),
+    ///     Ok(&tinystr!(3, "USD")));
+    /// assert!(matches!(
+    ///     TinyAsciiStr::<3>::try_from_raw(b"\0A\0"),
+    ///     Err(_)));
+    /// ```
+    pub const fn try_from_raw(raw: &[u8; N]) -> Result<&Self, TinyStrError> {
```

Comment:
I like `try_from_raw` because it makes sense alongside the existing subtag functions: `into_raw` and `from_raw_unchecked`.

Reply:
Also I think we won't need any other versions of this function, except perhaps a mutable version. We don't need a moving version because the types are `Copy`.

Reply:
Hmm, that's fair I guess. The weird thing about this is that it _returns_ a borrow, which is what I find notable

Reply:
Actually, come to think of it, these are all Copy types: is there a particular reason to want to be dealing in references at all?

Reply:
I'm looking at the use case in the other PR and rather skeptical of this.